src/map/ai/controllers/mob_controller.cpp

```diff
@@ -1412,3 +1412,26 @@ bool CMobController::IsSpellReady(float currentDistance)
 
     return (m_Tick >= m_LastMagicTime + std::chrono::milliseconds(PMob->getBigMobMod(MOBMOD_MAGIC_COOL) - bonusTime));
 }
+
+bool CMobController::HandleDrawin(float currentDistance)
+{
+    // get specific draw-in settings that can be set on any given mob otherwise use default values
+    uint32 drawInTriggerDistance = PMob->getMobMod(MOBMOD_DRAW_IN_TRIGGER_DIST) > 0 ? PMob->getMobMod(MOBMOD_DRAW_IN_TRIGGER_DIST) : PMob->GetMeleeRange() * 2;
+    uint32 drawInMaxDistance     = PMob->getMobMod(MOBMOD_DRAW_IN_MAX_RANGE) > 0 ? PMob->getMobMod(MOBMOD_DRAW_IN_MAX_RANGE) : std::numeric_limits<int16>::max();
+    bool   includeAlliance       = (PMob->getMobMod(MOBMOD_DRAW_IN) & DRAWIN::DRAWIN_INCLUDE_ALLIANCE) > 0 ? true : false;
+    bool   toFrontOfMob          = (PMob->getMobMod(MOBMOD_DRAW_IN) & DRAWIN::DRAWIN_TO_FRONT_OF_MOB) > 0 ? true : false;
```

Comment:
Same here, you don't need to be doing all of these extra steps...

```cpp
const bool toFrontOfMob = PMob->getMobMod(MOBMOD_DRAW_IN) & DRAWIN::DRAWIN_TO_FRONT_OF_MOB;
```

Reply:
Also, if you're not going to change the value of the variable later, make it const.

Reply:
> Also, if you're not going to change the value of the variable later, make it const.

Yep, I have fixed both of these, thanks for comment.